NitroxClient/Unity/Helper/GUISkinUtils.cs

```diff
@@ -0,0 +1,139 @@
+
+            copy.settings.cursorColor = baseSkin.settings.cursorColor;
+            copy.settings.cursorFlashSpeed = baseSkin.settings.cursorFlashSpeed;
+            copy.settings.doubleClickSelectsWord = baseSkin.settings.doubleClickSelectsWord;
+            copy.settings.tripleClickSelectsLine = baseSkin.settings.tripleClickSelectsLine;
+            copy.settings.selectionColor = baseSkin.settings.selectionColor;
+
+            // TODO: Create identical copy of font.
+            copy.font = baseSkin.font; // Gives a giberish font: copy.font = UnityEngine.Object.Instantiate(baseSkin.font);
+
+            return copy;
+        }
+
+        public static GUISkin RegisterDerived(string name, GUISkin baseSkin = null)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+            if (guiSkins.ContainsKey(name))
+            {
+                throw new ArgumentException("Name of new GUISkin already exists.", nameof(name));
+            }
+            
+            GUISkin newSkin = CreateDerived(baseSkin, name);
+            guiSkins.Add(name, newSkin);
+            return newSkin;
+        }
+
+        /// <summary>
+        /// Creates a new default Unity skin if there are no existing skins with the given <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">Name for the new skin.</param>
+        /// <param name="skinInitializer">Optional skin initializer.</param>
+        /// <param name="baseSkin">Optional base skin to copy from.</param>
+        /// <returns>New or cached skin.</returns>
+        public static GUISkin RegisterDerivedOnce(string name, Action<GUISkin> skinInitializer = null, GUISkin baseSkin = null)
+        {
+            if (!guiSkins.ContainsKey(name))
+            {
+                GUISkin newSkin = RegisterDerived(name, baseSkin);
+                skinInitializer?.Invoke(newSkin);
+            }
+
+            return guiSkins[name];
+        }
+
+        /// <summary>
+        /// Switches the active skin until after the <paramref name="onGui"/> action.
+        /// </summary>
+        /// <param name="skin">Skin to switch to.</param>
+        /// <param name="onGui">Render function to run.</param>
+        public static void SwitchSkin(GUISkin skin, Action onGui)
+        {
+            if (onGui == null)
+            {
+                throw new ArgumentNullException(nameof(onGui));
+            }
+
+            GUISkin prevSkin = GUI.skin;
+            GUI.skin = skin;
+            onGui();
+            GUI.skin = prevSkin;
+        }
+
+        public static void SetCustomStyle(this GUISkin skin, string name, GUIStyle baseStyle, Action<GUIStyle> modify)
+        {
+            GUIStyle style = new GUIStyle(baseStyle);
+            style.name = name;
+            modify(style);
+
+            bool found = false;
+            for (int i = 0; i < skin.customStyles.Length; i++)
```

Comment:
Personally not a fan of loops accessing something in an iterable in C#. At least this can be mitigated with `Array.FindIndex(skin.customStyles, item => item.name == style.name);`.